src/libraries/Microsoft.PowerFx.Core/Functions/Delegation/DelegationMetadata/FilterOpMetadata.cs

```diff
@@ -2,43 +2,90 @@
 // Licensed under the MIT license.
 
 using System.Collections.Generic;
+using Microsoft.PowerFx.Core.Entities;
 using Microsoft.PowerFx.Core.Types;
 using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Types;
 
 namespace Microsoft.PowerFx.Core.Functions.Delegation.DelegationMetadata
 {
     // Defines filter operation metadata.
     internal sealed class FilterOpMetadata : OperationCapabilityMetadata
     {
         private readonly Dictionary<DPath, DelegationCapability> _columnCapabilities;
+
         private readonly Dictionary<DPath, DelegationCapability> _columnRestrictions;
+
         private readonly DelegationCapability? _filterFunctionsSupportedByTable;
 
         // Filter functions supported at the table level.
         // If no capability at column level specified then this would be the default filter functionality supported by column.
         private readonly DelegationCapability _defaultCapabilities;
 
-        public FilterOpMetadata(
-            DType tableSchema,
-            Dictionary<DPath, DelegationCapability> columnRestrictions,
-            Dictionary<DPath, DelegationCapability> columnCapabilities,
-            DelegationCapability filterFunctionsSupportedByAllColumns,
-            DelegationCapability? filterFunctionsSupportedByTable)
+        private readonly TableDelegationInfo _delegationInfo;
+
+        public FilterOpMetadata(DType tableSchema, Dictionary<DPath, DelegationCapability> columnRestrictions, Dictionary<DPath, DelegationCapability> columnCapabilities, DelegationCapability filterFunctionsSupportedByAllColumns, DelegationCapability? filterFunctionsSupportedByTable)
             : base(tableSchema)
         {
             Contracts.AssertValue(columnRestrictions);
             Contracts.AssertValue(columnCapabilities);
 
             _columnCapabilities = columnCapabilities;
             _columnRestrictions = columnRestrictions;
+            _delegationInfo = null;
             _filterFunctionsSupportedByTable = filterFunctionsSupportedByTable;
             _defaultCapabilities = filterFunctionsSupportedByAllColumns;
+
             if (_filterFunctionsSupportedByTable != null)
             {
                 _defaultCapabilities = filterFunctionsSupportedByAllColumns | DelegationCapability.Filter;
             }
         }
 
+        public FilterOpMetadata(AggregateType schema, TableDelegationInfo delegationInfo)
```

Comment:
>FilterOpMetadata [](http://example.com/codeflow?start=15&length=16)

Original ctor is for interanl usage - which allows eagerly expanding capabilities. 
New ctor is for public usage, which allows lazy getting per-column capabilities.